### PR TITLE
Migrate cloudProviderId

### LINKED ISSUE #196: Migrate e2etest to an other cloudProviderId
### What happened

Conformance test failed

### Step to reproduce

launch conformance test

### Expected to happen

We migrate from osc:// to aws:/// with  (https://github.com/outscale-dev/cloud-provider-osc/blob/e565af1d8659f539bc70d5ece967d105196db4aa/cloud-controller-manager/osc/instances.go#L68)
Template are migrate, testenv are migrate.
We have also to migrate e2etest to the new cloudProviderId.

### Add anything

None

### cluster-api output

None

### Environment

```markdown
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version: 
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind bug     

**What this PR does / why we need it**:
Migrate cloudProviderId from osc:// to aws:///
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #196 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by outscale-vbr** [comment]
@outscale-hmi @outscale-mdr Are you ok with latest changes ?

File: example/cluster-machine-template-simple.yaml [hunk 1]
```diff
@@ -277,7 +277,7 @@ spec:
           name: "{{ ds.meta_data.local_hostname }}"
           kubeletExtraArgs:
             cloud-provider: external
-            provider-id: aws:///'{{ ds.meta_data.instance_id }}'
+            provider-id: aws://eu-west-2a/'{{ ds.meta_data.instance_id }}'
```
Comment (outscale-mdr) [inline example/cluster-machine-template-simple.yaml:280]: "eu-west-2a", it should be retrieve from the metadata.
The following command will give you the subregion:
```
curl  http://169.254.169.254/latest/meta-data/placement/availability-zone
```

You should test if ```{{ ds.meta_data.placement.availability_zone }}``` gives you what you want

File: test/e2e/data/infrastructure-outscale/cluster-template-kcp-remediation.yaml [hunk 2]
```diff
@@ -129,7 +129,7 @@ spec:
           name: "{{ ds.meta_data.local_hostname }}"
           kubeletExtraArgs:
             cloud-provider: external
-            provider-id: osc://'{{ ds.meta_data.instance_id }}'
+            provider-id: aws://${OSC_SUBREGION_NAME}/'{{ ds.meta_data.instance_id }}'
```
Comment (outscale-mdr) [inline test/e2e/data/infrastructure-outscale/cluster-template-kcp-remediation.yaml:132]: Same takes what the metadata server gives you

File: test/e2e/suite_test.go [hunk 3]
```diff
@@ -131,7 +131,14 @@ func getK8sClient() {
...
 	Expect(cfg).NotTo(BeNil())
-	k8sManager, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{})
+	retryPeriod := 30 * time.Second
+	leaseDuration := 80 * time.Second
+	renewDeadline := 20 * time.Second
+	k8sManager, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		LeaseDuration: &leaseDuration,
+		RenewDeadline: &renewDeadline,
+		RetryPeriod:   &retryPeriod,
+	})
```
Comment (outscale-mdr) [inline test/e2e/suite_test.go:141]: What is it for ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: test/e2e/suite_test.go (hunk 3 above)
Comment (outscale-mdr) [reply test/e2e/suite_test.go:141]: Ok thank you but now are you sure the controller will no be a master ?

File: test/e2e/data/infrastructure-outscale/cluster-template-kcp-remediation.yaml (hunk 2 above)
Comment (outscale-vbr) [reply test/e2e/data/infrastructure-outscale/cluster-template-kcp-remediation.yaml]: @outscale-mdr CI passed and validation on cloudgouv passed

File: example/cluster-machine-template-simple.yaml (hunk 1 above)
Comment (outscale-vbr) [reply example/cluster-machine-template-simple.yaml]: Ok done thanks.

**Pull request merged**